src/lib/core/projection/projection.ts

```diff
@@ -0,0 +1,56 @@
+import {Injectable, Directive, ModuleWithProviders, NgModule, ElementRef} from '@angular/core';
+
+
+@Directive({
+  selector: 'md-host'
+})
+export class MdProjectionHostDirective {
+  constructor(public ref: ElementRef) {}
+}
+
+
+@Injectable()
+export class MdProjectionService {
+  private _replaceWith(toReplaceEl: HTMLElement, otherEl: HTMLElement) {
+    toReplaceEl.parentElement.replaceChild(otherEl, toReplaceEl);
+  }
+
+  /**
+   * Project an element into a host element.
```

Comment:
This comment needs some expansion; the key thing to communicate is that it's *only* the given element that is projected and *not* its content (children). Also worth mention when you might want to use this.